cpu/stm32f1/periph/adc.c

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (C) 2015 Engineering-Spirit
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     cpu_stm32f1
+ * @{
+ *
+ * @file
+ * @brief       Low-level ADC driver implementation
+ *
+ * @author      Nick van IJzendoorn <nijzendoorn@engineering-spirit.nl>
+ *
+ * @}
+ */
+
+#include "cpu.h"
+#include "periph/adc.h"
+#include "periph_conf.h"
+
+/* guard in case that no ADC device is defined */
+#if ADC_NUMOF
+
+static bool _adc_get_pin(int channel, GPIO_TypeDef * *port, int *pin)
+{
+    if (channel > 17) {
+        return false;
+    }
+
+    if (channel == 8 || channel == 9)
+    {
+        *port = GPIOB;
+        *pin = channel - 8;
+    }
+    else if (channel < 8) {
+        *port = GPIOA;
+        *pin = channel;
+    }
+    else if (channel > 9) {
+        *port = GPIOC;
+        *pin = channel - 10;
```

Comment:
Wow, that looks weird :-) . Could you point to the regarding section in the reference manual so I don't need to search for it? Would be nice of you